src/lib/plugin/index.ts

```diff
@@ -145,378 +132,35 @@ export type RuntimeContributions<C extends {} = any> = {
-          part: 'workflow',
-          plugin: driver.name,
-        })
-        workflowComponent = driver.loadWorkflowPlugin(layout)
-      } catch (error) {
-        fatal(
-          stripIndent`
-          Error while trying to load the workflow component of plugin "${driver.name}":
-          
-          ${error}
-        `
-        )
-      }
-      return { name: driver.name, hooks: workflowComponent }
-    })
-
-  return workflowHooks
-}
-
-/**
- * Load all runtime plugins that are installed into the project.
- */
-export async function loadAllRuntimePluginsFromPackageJson(): Promise<
-  RuntimeContributions[]
-> {
-  const plugins = await loadAllFromPackageJson()
-  return __doLoadAllRuntimePluginsFromPackageJson(plugins)
-}
-
-/**
- * Load all runtime plugins that are installed into the project.
- */
-export function loadAllRuntimePluginsFromPackageJsonSync(): RuntimeContributions[] {
-  const plugins = loadAllFromPackageJsonSync()
-  return __doLoadAllRuntimePluginsFromPackageJson(plugins)
-}
-
-export async function loadAllTestingPluginsFromPackageJson(): Promise<
-  TestingContributions[]
-> {
-  const plugins = await loadAllFromPackageJson()
-
-  const testingPlugins = plugins
-    .filter(driver => driver.extendsTesting)
-    .map(driver => {
-      let testingContributions: TestingContributions
-      try {
-        pluginSystemLogger.trace('load', {
-          part: 'testing',
-          plugin: driver.name,
-        })
-        testingContributions = driver.loadTestingPlugin()! // guaranteed by above filter
-      } catch (error) {
-        fatal(
-          stripIndent`
-          Error while trying to load the testing component of plugin "${driver.name}":
-          
-          ${error}
-        `
-        )
-      }
-      return testingContributions
-    })
-
-  return testingPlugins
-}
-
-/**
- * Logic shared between sync/async variants.
- */
-function __doLoadAllRuntimePluginsFromPackageJson(
-  plugins: Driver[]
-): RuntimeContributions[] {
-  const runtimePlugins = plugins
-    .filter(driver => driver.extendsRuntime)
-    .map(driver => {
-      let runtimeContributions: RuntimeContributions
-      try {
-        pluginSystemLogger.trace('load', {
-          part: 'runtime',
-          plugin: driver.name,
-        })
-        runtimeContributions = driver.loadRuntimePlugin()! // guaranteed by above filter
-      } catch (error) {
-        fatal(
-          stripIndent`
-          Error while trying to load the runtime component of plugin "${driver.name}":
-          
-          ${error}
-        `
-        )
-      }
-      return runtimeContributions
-    })
-
-  return runtimePlugins
-}
+  packageManager: PackageManager.PackageManager
+}
+
+// prettier-ignore
+export type WorktimePlugin = (hooks: WorktimeHooks, lens: WorktimeLens) => void
+// prettier-ignore
+export type RuntimePlugin = (lens: Lens) => RuntimeContributions
+// prettier-ignore
+export type TesttimePlugin = (lens: Lens) => TesttimeContributions
```

Comment:
Can we maybe move `hooks` as second parameter to keep the `lens` param position consistent across the different `dimensions` ?